tests/ice40/memory.ys

```diff
@@ -0,0 +1,7 @@
+proc
+memory
+equiv_opt -map ../../techlibs/ice40/cells_sim.v synth_ice40
+synth_ice40
+select -assert-count 8 t:SB_DFF
+select -assert-count 512 t:SB_DFFE
```

Comment:
This should map to a memory, but is creating 500+ flops instead. How come?

Reply:
If use just synth_ice40 we have this result:
   Number of cells:                  1
     SB_RAM40_4K                     1

but equiv_opt reports: ERROR: Gold module contains memories or procresses. Run 'memory' or 'proc' respectively.

But when use this commands:
read_verilog memory.v
proc
memory
synth_ice40

I have this result (like in current test case):

     SB_DFF                          8
     SB_DFFE                       512
     SB_LUT4                       464


When use the same commands without proc, the result will be:
     $mem                            1
     SB_DFF                         23

Maybe there is a bug but without memory command I cant use equiv_opt, and with it I don't have an SB_RAM40_4K.

Reply:
Running `proc` and `memory` before `synth_ice40` blocks it from inferring a RAM.
However, even if you fix this issue and others, it looks like there's an issue with equivalency checking memories, so maybe we should resort to just `select -assert-count` and simulation for this particular test.

Reply:
done